src/hotspot/share/opto/node.cpp

```diff
@@ -1696,6 +1696,670 @@ bool Node::add_to_worklist(Node* n, Node_List* worklist, Arena* old_arena, Vecto
+    return;
+  }
+  Info* info = _bfs->find_info(n);
+  if (info == nullptr || !info->is_marked()) {
+    return;
+  }
+
+  const Type* t = n->bottom_type();
+  switch (t->category()) {
+    case Type::Category::Data:
+      st->print("\u001b[34m");
+      break;
+    case Type::Category::Memory:
+      st->print("\u001b[32m");
+      break;
+    case Type::Category::Mixed:
+      st->print("\u001b[35m");
+      break;
+    case Type::Category::Control:
+      st->print("\u001b[31m");
+      break;
+    case Type::Category::Other:
+      st->print("\u001b[33m");
+      break;
+    case Type::Category::Undef:
+      n->dump();
+      assert(false, "category undef ??");
+      break;
+    default:
+      n->dump();
+      assert(false, "not covered");
+      break;
+  }
+}
+
+void PrintBFS::DumpConfigColored::post_dump(outputStream* st) {
+  if (!_bfs->_use_color) {
+    return;
+  }
+  st->print("\u001b[0m"); // white
+}
+
+Node* PrintBFS::old_node(Node* n) {
+  Compile* C = Compile::current();
+  if (C->matcher() == nullptr || !C->node_arena()->contains(n)) {
+    return (Node*)nullptr;
+  } else {
+    return C->matcher()->find_old_node(n);
+  }
+}
+
+void PrintBFS::print_node_idx(Node* n) {
+  Compile* C = Compile::current();
+  char buf[30];
+  if (n == nullptr) {
+    sprintf(buf,"_");           // null
+  } else if (C->node_arena()->contains(n)) {
+    sprintf(buf, "%d", n->_idx);  // new node
+  } else {
+    sprintf(buf, "o%d", n->_idx); // old node
+  }
+  tty->print("%6s", buf);
+}
+
+void PrintBFS::print_node_block(Node* n) {
+  Compile* C = Compile::current();
+  Block* b = C->node_arena()->contains(n)
+             ? C->cfg()->get_block_for_node(n)
+             : nullptr; // guard against old nodes
+  if (b == nullptr) {
+    tty->print("     _");
+    tty->print("     _");
+    tty->print("   _");
+  } else {
+    print_node_idx(b->head());
+    if (b->_idom) {
+      print_node_idx(b->_idom->head());
+    } else {
+      tty->print("     _");
+    }
+    tty->print("%4d", b->_dom_depth);
+  }
+}
+
+// filter, and add to worklist, add info, note traversal edges
+void PrintBFS::maybe_traverse(Node* src, Node* dst) {
+  if (dst != nullptr &&
+     (filter_category(dst, _filter_visit) ||
+      filter_category(dst, _filter_boundary) ||
+      dst == _start)) { // correct category or start?
+    if (find_info(dst) == nullptr) {
+      // never visited - set up info
+      _worklist.push(dst);
+      int d = 0;
+      if (dst != _start) {
+        d = find_info(src)->distance() + 1;
+      }
+      make_info(dst, d);
+    }
+    if (src != dst) {
+      // traversal edges useful during select
+      find_info(dst)->edge_bwd.push(src);
+    }
+  }
+}
+
+void PrintBFS::print_header() const {
+  tty->print("   d");                         // distance
+  if (_all_paths) {
+    tty->print(" apd");                       // all paths distance
+  }
+  if (_print_blocks) {
+    tty->print(" [head  idom  d]");           // block
```

Comment:
Perhaps rename the third block column `d` to something more descriptive (`depth` or similar) for clarity, that would also break the ambiguity with `d` which is already used for "distance".